### PR TITLE
Feature/fluxification

### PR DESCRIPTION
**Pull request by andre-merzky**
This PR adds preliminary FLUX support: scheduler, resource manager, and launch method.

### CODE REVIEW

File: src/radical/pilot/agent/scheduler/flux.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        AgentSchedulingComponent.__init__(self, cfg, session)
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _configure(self):
+
+        import flux
+
+        flux_url   = self._cfg['lrms_info']['lm_info']['flux_env']['FLUX_URI']
```
Comment (andre-merzky) [inline src/radical/pilot/agent/scheduler/flux.py:40]: `lrms` -> name changed

File: src/radical/pilot/agent/scheduler/flux.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        from flux import job as flux_job
+
+        self.advance(units, rps.AGENT_SCHEDULING, publish=True, push=False)
+
+        for unit in units:
+
+          # # FIXME: transfer from executor
+          # self._cu_environment = self._populate_cu_environment()
+
+            jd  = json.dumps(ru.read_json('/home/merzky/projects/flux/spec.json'))
```
Comment (andre-merzky) [inline src/radical/pilot/agent/scheduler/flux.py:73]: add example, maybe initial translation, check GPUs

File: src/radical/pilot/agent/executing/flux.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+    #
+    def _listen(self):
+
+        flux_handle = None
+
+        try:
+            # thread local initialization
+            import flux
+
+            flux_url    = self._cfg['lrms_info']['lm_info']\
```
Comment (andre-merzky) [inline src/radical/pilot/agent/executing/flux.py:130]: `lrms`

File: src/radical/pilot/agent/scheduler/flux.py (hunk 2 above)
Comment (mtitov) [reply src/radical/pilot/agent/scheduler/flux.py:73]: just to ping this comment (in case it is still relevant)

Comment (andre-merzky) [reply src/radical/pilot/agent/scheduler/flux.py:73]: thanks for the ping, will do.

**Pull request merged**